src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.cs

```diff
@@ -1467,7 +1439,8 @@ public static Tensor<T> Reshape<T>(this Tensor<T> tensor, ReadOnlySpan<nint> len
 
             nint[] newLengths = lengths.ToArray();
             // Calculate wildcard info.
-            if (lengths.Contains(-1))
+            int wildcardIndex = lengths.IndexOf(-1);
+            if (wildcardIndex != -1)
```

Comment:
Should we make it `>= 0` instead? I know currently we return -1 if not found but I think in general the IndexOf contract is to return any negative number when not found.

Reply:
I'm happy to make the switch. I pulled the -1 from this doc, https://learn.microsoft.com/en-us/dotnet/api/system.array.indexof?view=net-9.0#system-array-indexof-1(-0()-0), which does specifically say it uses -1 if its not found.

I'll defer to your experience in this area as I don't want to introduce a potential future bug so if you think its better to change ill change it.

Reply:
That is fine to keep `-1` check.